### DISCUSSION TITLE
Support for azure function requests in application insights, like "OpenTelemetry.Instrumentation.AspNetCore" does for asp .net core app

### DISCUSSION THREAD

**Discussion by Cheng811** [open-telemetry/opentelemetry-dotnet#4382] [Q&A]
We are try to use opentelemetry trace in a project of azure function. But we find that we could not write requests table in application insights like we did in asp .net core project. The package of "OpenTelemetry.Instrumentation.AspNetCore" provides the function of "AddAspNetCoreInstrumentation" as the code shows. But it does not works in azure function. Is there any support on it for azure function?
`        services.AddOpenTelemetryTracing(builder =>
        {
            builder
            .SetResourceBuilder(openTelemetryResourceBuilder)
            .AddAspNetCoreInstrumentation() // Auto-instrument http incoming requests
            .AddHttpClientInstrumentation() // Auto-instrument http outgoing dependencies like calls triggered by HttpClient
            .AddConsoleExporter()
            .AddAzureMonitorTraceExporter(option => option.ConnectionString = $"InstrumentationKey={instrumentationKey ?? "00000000-0000-0000-0000-000000000000"}");
        });`

**Comment by gitmadness** [comment]
I know there's an old example at https://github.com/cijothomas/FunctionsOpenTelemetry/blob/master/FunctionsOpenTelemetry/Startup.cs but I wonder if there is an up-to-date example.

**Reply by cijothomas** [reply]
^ This repo is no longer accurate and should not be relied upon.

**Comment by cijothomas** [comment]
_But it does not works in azure function. Is there any support on it for azure function?_

Asp.Net Core instrumentation should work everywhere Asp.Net Core is supported. Azure Functions is probably not using Asp.Net Core internally. I'd recommend to ask this in Azure Functions repo to get an official answer.

(It used to work sometime, but it was relying on Functions internally using asp.net core)
